feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/base_vrf_selection_test.go

```diff
@@ -176,51 +174,64 @@ func configForwardingPolicy() *oc.NetworkInstance_PolicyForwarding {
 	ni := d.GetOrCreateNetworkInstance(*deviations.DefaultNetworkInstance)
 	ipv4Address := "0.0.0.0/0"
 	ipv6Address := "::/0"
-	ipv6dest := "2001:db8::1/128"
+	icmp6 := 58
```

Comment:
why do we need icmp6 here? the test description only asks for matching on protocol ipv4 or ipv6 or ipinip and dscp set.

Reply:
IxiaPort1--------DUTPORT1------------DUTPORT2----VLAN10
                                                                                   ----VLAN20
 
The TC wants to ensure expected IPv4/IPv6  traffic (based on sub TC) is allowed which means others should get rejected. The script configures IPv4 and IPv6 addresses in all interfaces 1 at ingress and 2 egress sub-interfaces.

 If policy has only IPv4 terms, it rejects any IPv6 packet including ICMP6 , this stops NDP between IxiaPort1 and DUTPORT1. To allow NDP, ICMP6 is allowed, you see similiar code for all policies.  To allow ICMP6/NDP discard is set to False.

Reply:
@azubaidhar this looks like vendor specific implementation for traffic redirection. 
as per `https://github.com/openconfig/public/blob/master/release/models/policy-forwarding/openconfig-pf-forwarding-policies.yang#L200`
```
        enum VRF_SELECTION_POLICY {
          description
            "The policy is used only to classify incoming packets into corresponding
            network instances.";
        }
      }
```
and as per `https://github.com/openconfig/public/blob/master/release/models/policy-forwarding/openconfig-pf-interfaces.yang#L153`

```
      description
        "Apply the specific VRF selection policy on the interface. The referenced policy
        MUST be of type VRF_SELECTION_POLICY. The VRF selection policy may coexist with
        a policy-forwarding policy.
        The policy specified in this leaf is used to specifically choose the L3VRF network
        instance that is used for specific input criteria of packets.";
    }
```

If the traffic is matched for protocol ipv4, we should only redirect ipv4 traffic to a specific vrf.

Please take a look at the below comment  by @robshakir  _"Given this, it is probably even worth us extending the test case to cover that IP forwarding that doesn't match the VRF selection policy is forwarded."_:
https://github.com/openconfig/featureprofiles/pull/847#discussion_r1086184777
Which suggests that unmatched traffic should not be discarded .